rfc/image-discoverability.md

```diff
@@ -0,0 +1,127 @@
+    gcpimage --> gcpupload{GCP image upload}
+    gcpimage --> gcpmeasure{measure}
+    awsimage --> awsmeasure{measure}
+    azureimage --> azuremeasure{measure}
+    awsupload --> awsimage1[ami-123]
+    awsupload --> awsimage2[ami-456]
+    awsupload --> awsimage3[ami-789]
+    azureupload --> azureimage1[azure-cvm-123]
+    azureupload --> azureimage2[azure-trusted-launch-123]
+    gcpupload --> gcpimage1[gcp-image-123]
+    awsmeasure --> awsmeasurements[measurements.yml]
+    azuremeasure --> azuremeasurements[measurements.yml]
+    gcpmeasure --> gcpmeasurements[measurements.yml]
+    awsimage1 --> lut{Upload image lookup table to S3<br>key: <code>v2.2.0-cc0de5c</code><br>value: lookup table for image references specific to each CSP}
+    awsimage2 --> lut
+    awsimage3 --> lut
+    azureimage1 --> lut
+    azureimage2 --> lut
+    gcpimage1 --> lut
+    awsmeasurements --> s3measurements{"Upload image measurements to S3<br><code>aws/v2.2.0-cc0de5c/measurements.yaml<br>azure/v2.2.0-cc0de5c/measurements.yaml<br>gcp/v2.2.0-cc0de5c/measurements.yaml<br></code>"}
+    azuremeasurements --> s3measurements
+    gcpmeasurements --> s3measurements
+```
+
+## Inputs and outputs of the build pipeline
+
+The build pipeline takes as inputs:
+
+- a version number that is one of
+  - a release version number (e.g. `v2.2.0`) for release images
+  - a pseudo-version number (e.g. `v2.3.0-pre-debug`) for development images
+  - a pseudo-version number (e.g. `v2.3.0-pre-branch-name`) for branch images
+- a commit hash of the Constellation monorepo that is used to build the images (e.g. `cc0de5c68d41f31dd0b284d574f137e0b0ad106b`)
+
+To identify images belonging to one invocation of the build pipeline, the pipeline uses a unique identifier for the set of images, referred to as `image version uid` that combines the version number and the commit hash (e.g. `v2.2.0-cc0de5c68d41f31dd0b284d574f137e0b0ad106b`).
+
+The build pipeline produces as outputs:
+
+- a raw OS image for every CSP
+- a set of measurements for each raw OS image
+- one or more images uploaded to each CSP (e.g. AWS AMIs, Azure images, GCP images)
+- a lookup table that maps the `image version uid` to the CSP-specific image references
+
+The lookup table is uploaded to S3 and is used to identify the images that belong to a given `image version uid`.
+Measurements are uploaded to S3 and can be looked up for each cloud service provider and `image version uid`.
+
+## Image lookup table
+
+The image lookup table is a JSON file that maps the `image version uid` to the CSP-specific image references. It uses the `image version uid` as file name.
+
+```
+s3://<BUCKET-NAME>/images/<IMAGE-VERSION-UID>.json
+```
+
+```json
+{
+  "aws": {
+    "us-east-1": "ami-123",
+    "us-west-2": "ami-456",
+    "eu-west-1": "ami-789"
+  },
+  "azure": {
+    "cvm": "azure-cvm-123",
+    "trustedlaunch": "azure-trusted-launch-123"
+  },
+  "gcp": "gcp-image-123",
+  "qemu": "https://cdn.edgeless.systems/.../qemu.raw"
```

Comment:
don't we want to upload the qemu image as compressed qcow2?

Reply:
The root file system itself is a compressed squashfs since switching to mkosi so I used raw images on QEMU for my own testing as there is not a lot to gain. It may still be a good idea to stay with QCOW for consistency.

Reply:
Understood. Then I've got no opinion on this.